nornir/core/configuration.py

```diff
@@ -16,7 +16,7 @@
 T = TypeVar("T")
 
 
-class Parameter:
+class Parameter(Generic[T]):
```

Comment:
This looked like the recommended way to enforce type consistency i.e. that a Parameter that was a 'str' was consistently required to be a 'str').

There were some mypy issues until I did this.